### PR TITLE
Add load balancing configuration for edge gateway

### PR DESCRIPTION
**Pull request by Didainius**
As part of load balancer introduction ( https://github.com/terraform-providers/terraform-provider-vcd/issues/223) this PR adds capability to set general load balancing settings (such as enable/disable load balancing)

### CODE REVIEW

File: vcd/resource_vcd_edgegateway.go [hunk 1]
```diff
@@ -145,12 +168,107 @@ func resourceVcdEdgeGatewayCreate(d *schema.ResourceData, meta interface{}) erro
...
+			return err
+		}
+	}
+
+	log.Printf("[TRACE] edge gateway read completed: %#v", edgeGateway.EdgeGateway)
+	return nil
+}
+
+// resourceVcdEdgeGatewayUpdate updates an edge gateway from a resource definition
+func resourceVcdEdgeGatewayUpdate(d *schema.ResourceData, meta interface{}) error {
```
Comment (lvirbalas) [inline vcd/resource_vcd_edgegateway.go]: Does this function update only LB general params of EG currently? If yes, a fat comment about such behavior is needed.

File: vcd/resource_vcd_edgegateway.go [hunk 2]
```diff
@@ -196,44 +314,45 @@ func setEdgeGatewayValues(d *schema.ResourceData, egw govcd.EdgeGateway) error {
...
+		return fmt.Errorf("unable to update general load balancer settings: %s", err)
 	}
 
-	err = edgeGateway.Delete(true, true)
+	return nil
+}
 
-	log.Printf("[TRACE] edge gateway deletion completed\n")
-	return err
+// isEdgeGatewayLbConfigured checks if any of load balancer related settings are set
```
Comment (lvirbalas) [inline vcd/resource_vcd_edgegateway.go]: I suggest adding that this only applies to LB **general parameters**.

File: website/docs/r/edgegateway.html.markdown [hunk 3]
```diff
@@ -60,5 +63,17 @@ The following arguments are supported:
 * `advanced` - (Optional) True if the gateway uses advanced networking. Default is `true`.
 * `ha_enabled` - (Optional) Enable high availability on this edge gateway. Default is `false`.
 * `distributed_routing` - (Optional) If advanced networking enabled, also enable distributed routing. Default is `false`.
+* `lb_enabled` - (Optional) Enable load balancing. Default is `false`.
```
Comment (lvirbalas) [inline website/docs/r/edgegateway.html.markdown]: Currently these new optional LB parameters in resource definition don't have default values, but we are saying three of them are false by default. I suspect it becomes false if undefined? What about actually adding a default = false, so the behavior has less assumptions? Or would that introduce something unwanted?

Comment (Didainius) [reply website/docs/r/edgegateway.html.markdown]: I experimented with that. One thing I didn't like:
If I set `default` I can't make the check if value `is set` in config or is it a `default`. Then it always tries to configure load balancer (with default values if not set).
The actual default configuration at least on 9.5 (but I think others as well) after creation is enabled=false, lb_acceleration_enabled=false, lb_logging_enabled=false, lb_loglevel='info'.
I could set such defaults, then after initial read it wouldn't complain as these values coincide with defaults, but if in some future version of vCD the defaults change then I would start overriding them.

File: vcd/resource_vcd_edgegateway.go (hunk 1 above)
Comment (dataclouder) [inline vcd/resource_vcd_edgegateway.go]: We have an issue about implementing Update for EGW. Please add a note saying that this Update is only dealing with LB items so far.

Comment (vbauzysvmware) [reply vcd/resource_vcd_edgegateway.go]: Then maybe we need create Issue to tackle full edge gtw update?

Comment (dataclouder) [reply vcd/resource_vcd_edgegateway.go]: we have one https://github.com/vmware/go-vcloud-director/issues/202

Comment (Didainius) [reply vcd/resource_vcd_edgegateway.go]: I changed the comment to explicitly state that it is general load balancer params. Also resource schema hints it quite well with only `lb_*` fields not having `ForceNew: true`

File: vcd/resource_vcd_edgegateway.go (hunk 2 above)
Comment (Didainius) [reply vcd/resource_vcd_edgegateway.go]: Getting rid of it as we are going to set default load balancer settings always. This check doesn't help anymore when defaults are used.

File: website/docs/r/edgegateway.html.markdown (hunk 3 above)
Comment (Didainius) [reply website/docs/r/edgegateway.html.markdown]: Changed these fields to always have default values (same as edge gateway currently gets during provisioning):
enabled=false
lb_acceleration_enabled=false
lb_logging_enabled=false
lb_loglevel='info'

The only scenario when they are ignored is when edge gateway is _not_ advanced.
It also prints a warning if someone is trying to enable it on `simple` edge gateway during `updates procedures and the website docs have a note now.

File: website/docs/r/edgegateway.html.markdown [hunk 4]
```diff
@@ -17,8 +17,10 @@ Supported in provider *v2.4+*
 You must use `System Adminstrator` account in `provider` configuration
 and then provide `org` and `vdc` arguments for edge gateway to work.
 
-~> **Note:** To enable load balancing capabilities the edge gateway must be `advanced`. Refer to
-[official vCloud Director documentation](https://docs.vmware.com/en/vCloud-Director/9.7/com.vmware.vcloud.tenantportal.doc/GUID-7E082E77-B459-4CE7-806D-2769F7CB5624.html) for more information.
+~> **Note:** Load balancing capabilities will work only when edge gateway is `advanced`. Load
```
[the remark this answers is not in the document]
Comment (Didainius) [reply website/docs/r/edgegateway.html.markdown]: The `make website` handles it. Link is active and working.

[the remark this answers is not in the document]
Comment (dataclouder) [reply website/docs/r/edgegateway.html.markdown]: OK, then. It's only my IDE preview that complains

**Pull request merged**